src/Microsoft.Sbom.Api/Workflows/Helpers/PackageArrayGenerator.cs

```diff
@@ -21,53 +22,56 @@ public class PackageArrayGenerator : IJsonArrayGenerator<PackageArrayGenerator>
 {
     private readonly ILogger log;
 
-    private readonly ISbomConfigProvider sbomConfigs;
-
     private readonly IEnumerable<ISourcesProvider> sourcesProviders;
 
     private readonly IRecorder recorder;
 
+    private readonly ISbomConfigProvider sbomConfigs;
+
+    public ISbomConfig SbomConfig { get; set; }
+
+    public string SpdxManifestVersion { get; set; }
+
     public PackageArrayGenerator(
         ILogger log,
-        ISbomConfigProvider sbomConfigs,
         IEnumerable<ISourcesProvider> sourcesProviders,
-        IRecorder recorder)
+        IRecorder recorder,
+        ISbomConfigProvider sbomConfigs)
     {
         this.log = log ?? throw new ArgumentNullException(nameof(log));
-        this.sbomConfigs = sbomConfigs ?? throw new ArgumentNullException(nameof(sbomConfigs));
         this.sourcesProviders = sourcesProviders ?? throw new ArgumentNullException(nameof(sourcesProviders));
         this.recorder = recorder ?? throw new ArgumentNullException(nameof(recorder));
+        this.sbomConfigs = sbomConfigs ?? throw new ArgumentNullException(nameof(sbomConfigs));
     }
 
-    public async Task<IList<FileValidationResult>> GenerateAsync()
+    public async Task<GenerateResult> GenerateAsync()
     {
         using (recorder.TraceEvent(Events.PackagesGeneration))
         {
-            IList<FileValidationResult> totalErrors = new List<FileValidationResult>();
+            var totalErrors = new List<FileValidationResult>();
 
             var sourcesProvider = this.sourcesProviders
                 .FirstOrDefault(s => s.IsSupported(ProviderType.Packages));
 
             // Write the start of the array, if supported.
             IList<ISbomConfig> packagesArraySupportingConfigs = new List<ISbomConfig>();
-            foreach (var manifestInfo in sbomConfigs.GetManifestInfos())
-            {
-                var config = sbomConfigs.Get(manifestInfo);
-                if (config.MetadataBuilder.TryGetPackageArrayHeaderName(out var packagesArrayHeaderName))
-                {
-                    packagesArraySupportingConfigs.Add(config);
-                    config.JsonSerializer.StartJsonArray(packagesArrayHeaderName);
-                }
-            }
+            var serializationStrategy = JsonSerializationStrategyFactory.GetStrategy(SpdxManifestVersion);
+            serializationStrategy.AddToPackagesSupportingConfig(ref packagesArraySupportingConfigs, this.SbomConfig);
 
             var (jsonDocResults, errors) = sourcesProvider.Get(packagesArraySupportingConfigs);
 
-            // 6. Collect all the json elements and write to the serializer.
+            // 6. Collect all the json elements to be written to the serializer.
```

Comment:
Since we won't have 1 - 5, let's take advantage of this change to remove the `6. `